taichi/backends/cuda/cuda_profiler.cpp

```diff
@@ -0,0 +1,150 @@
+#include "taichi/backends/cuda/cuda_profiler.h"
+#include "taichi/backends/cuda/cuda_driver.h"
+#include "taichi/backends/cuda/cuda_context.h"
+
+TLANG_NAMESPACE_BEGIN
+
+CUDAProfiler::CUDAProfiler() {
+  TI_TRACE("CUDAProfiler::CUDAProfiler() ");
+}
+
+CUDAProfiler::~CUDAProfiler() {
+  if (profiler_config_.profiler_type == CUDA_KERNEL_PROFILER_CUPTI) {
+    end_profiling();
+    deinit_cupti();
+  }
+}
+
+bool CUDAProfiler::set_profiling_mode(KernelProfilerMode profiling_mode) {
+  bool enable = profiling_mode == KernelProfilerMode::enable |
+                profiling_mode == KernelProfilerMode::cuda_accurate |
+                profiling_mode == KernelProfilerMode::cuda_detailed;
+  if (!enable) {
+    return false;
+  }
+
+  CUDAKernalProfiler profiler_type =
+      (profiling_mode == KernelProfilerMode::cuda_accurate) ||
+              (profiling_mode == KernelProfilerMode::cuda_detailed)
+          ? CUDA_KERNEL_PROFILER_CUPTI
+          : CUDA_KERNEL_PROFILER_EVENT;
+  profiler_config_.profiling_mode = profiling_mode;
+
+  if (enable && profiler_type == CUDA_KERNEL_PROFILER_EVENT) {
```

Comment:
Since we return instantly when enable==false, it's probably safe to assume enable=true after line 22?

Reply:
thanks, the `enable` here is somewhat redundant. i have deleted it in commit e05ee990e753783bab8d57c1638e5d78893675f6